### ISSUE TITLE
[inductor][cuda] `align_random_eager=True` makes compiled `torch.randn` / `randn_like` return uniform(0,1) values instead of standard normal

### ISSUE THREAD

**Issue by BUAA-Wander** [pytorch/pytorch#193938]
## 🐛 Describe the bug

With `torch._inductor.config.align_random_eager = True`, compiled `torch.randn` and `torch.randn_like` on CUDA silently return uniform(0,1) values (min 0, max 1, mean ~0.5, std ~0.289) instead of standard normal. The unaligned path (`align_random_eager=False`) is correct, and CPU is unaffected. Root cause: in `torch/_inductor/lowering.py`, the `align_random_eager` branch of the rand/randn lowering always calls `ops.rand_eager(...)` (uniform), ignoring the `mode` argument, so `randn` never gets the normal transform.

## Reproducer

```python
import torch
import torch._inductor.config as ic
ic.align_random_eager = True

def fn():
    return torch.randn(4, 5000, device="cuda")

torch.manual_seed(1234)
x = torch.compile(fn)()
torch.cuda.synchronize()
print(x.min().item(), x.max().item(), x.mean().item(), x.std().item())
# (~0.0, ~1.0, ~0.5, ~0.29) -> uniform(0,1); eager randn is N(0,1)
```

## Environment

- torch 2.13.0+cu130 (CUDA)
- torch 2.15.0.dev20260817+cu130 (nightly, CUDA)
- CPU unaffected; `align_random_eager=False` unaffected.

## Expected behavior

`torch.randn` should keep the standard normal distribution even with `align_random_eager=True`; only the seed/offset alignment should change.

## Additional context

- Same behavior for `randn_like`; `rand` itself is correct (uniform, as expected).
- Related: #185600 (align_random_eager), #181136 (rand op order).

cc @pbelevich @chauhang @penguinwu @voznesenskym @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @ipiszy @kadeng @muchulee8 @amjames @aakhundov @coconutruben @jataylo

**Comment by Mukesh-SCS** [comment]
Hi @BUAA-Wander, I investigated this and opened a PR with a minimal fix.

The issue comes from `inductor_random()` applying the `align_random_eager` path to `randn` as well as `rand`. The FX pass only creates the eager-aligned seed/offset path for `rand`, so `randn` was incorrectly lowered through the uniform `rand_eager` kernel.

The fix restricts that lowering path to `mode == "rand"` and adds regression coverage for both `torch.randn` and `torch.randn_like`.

I’d appreciate it if you could take a look and confirm this matches the behavior you reported.
